catalyst-gateway/bin/src/service/api/legacy/v0/proposals_get.rs

```diff
@@ -0,0 +1,57 @@
+//! Implementation of the GET /setting endpoint
+use poem_openapi::{payload::Json, ApiResponse};
+
+use crate::service::common::responses::WithErrorResponses;
+
+/// Endpoint responses
+#[derive(ApiResponse)]
+pub(crate) enum Responses {
+    /// The default success response.
+    #[oai(status = 200)]
+    Ok(Json<Vec<dto::Proposal>>),
+}
+
+/// All responses
+pub(crate) type AllResponses = WithErrorResponses<Responses>;
+
+/// The service endpoint
+#[allow(clippy::unused_async)]
+pub(crate) async fn endpoint() -> AllResponses {
+    Responses::Ok(Json(Vec::new())).into()
+}
+
+/// The data transfer objects over HTTP
+pub(crate) mod dto {
+    use std::collections::HashMap;
+
+    use poem_openapi::Object;
+
+    #[allow(clippy::missing_docs_in_private_items)]
+    #[allow(clippy::struct_field_names)]
+    #[derive(Object, Default)]
+    pub(crate) struct Proposal {
+        internal_id: String,
+        chain_voteplan_id: String,
```

Comment:
The problem is I barely know the extact types of these because I tracked all the type from JS Norns, and that's the problem of JS. So if you have the type references to this, it'll be great, and I'll change this accordingly.

Reply:
Legacy endpoints do not need to have all the docs, but they do need to be added to the `catalyst-gateway/tests/.oapi-v3.spectral.yml` file so that they do not generate any warnings.
Those warning are going to become errors soon, which will break CI.

Reply:
From the `catalyst-gateway` folder, you can run `reset; just quick-schema-lint` which will build cat-gateway locally, and do a lint of the api.  Its the fastest way to check for OpenAPI issues.